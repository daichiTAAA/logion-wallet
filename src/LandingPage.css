.LandingPage {
    text-align: center;
}

.LandingPage .logo-slogan {
    margin-top: 30px;
    margin-left: 30px;
    display: flex;
    align-items: flex-end;
}

.LandingPage .logo-slogan p {
    margin: 0;
    margin-left: 25px;
}

.LandingPage .messages {
    margin-bottom: 50px;
}

.LandingPage .primary-message {
    font-family: 'Lexend Deca', sans-serif;
    font-size: 50px;
}

.LandingPage .secondary-message {
    font-size: 38px;
}
