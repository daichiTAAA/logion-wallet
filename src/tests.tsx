import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
