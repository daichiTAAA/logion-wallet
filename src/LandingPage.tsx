import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { Children } from './common/types/Helpers';

import Logo from './Logo';

import './LandingPage.css';

export interface Props {
    children: Children,
}

export default function LandingPage(props: Props) {

    return (
        <div className="LandingPage">
            <Container fluid>
                <Row>
                    <Col>
                        <div className="logo-slogan">
                            <Logo size={ 100 } />
                            <p>One blockchain <br/>to trust them all</p>
                        </div>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <div className="messages">
                            <p className="primary-message">The secure blockchain</p>
                            <p className="secondary-message">accessible in two steps</p>
                        </div>
                    </Col>
                </Row>
                { props.children }
            </Container>
        </div>
    );
}
