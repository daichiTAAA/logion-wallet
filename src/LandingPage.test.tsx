import { shallowRender } from './tests';

import LandingPage from './LandingPage';

test("renders", () => {
    const result = shallowRender(
        <LandingPage>
            Test
        </LandingPage>
    );
    expect(result).toMatchSnapshot();
});
