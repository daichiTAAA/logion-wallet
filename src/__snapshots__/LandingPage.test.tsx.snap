// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="LandingPage"
>
  <Container
    fluid={true}
  >
    <Row
      noGutters={false}
    >
      <Col>
        <div
          className="logo-slogan"
        >
          <Logo
            size={100}
          />
          <p>
            One blockchain 
            <br />
            to trust them all
          </p>
        </div>
      </Col>
    </Row>
    <Row
      noGutters={false}
    >
      <Col>
        <div
          className="messages"
        >
          <p
            className="primary-message"
          >
            The secure blockchain
          </p>
          <p
            className="secondary-message"
          >
            accessible in two steps
          </p>
        </div>
      </Col>
    </Row>
    Test
  </Container>
</div>
`;
